public/app/plugins/datasource/prometheus/components/PromExploreQueryEditor.tsx

```diff
@@ -26,34 +28,86 @@ export const PromExploreQueryEditor: FC<Props> = (props: Props) => {
     }
   }
 
+  function onQueryTypeChange(value: string) {
+    const { query, onChange } = props;
+    let nextQuery;
+    if (value === 'instant') {
+      nextQuery = { ...query, instant: true, runAll: false };
+    } else if (value === 'range') {
+      nextQuery = { ...query, instant: false, runAll: false };
+    } else {
+      nextQuery = { ...query, instant: true, runAll: true };
+    }
+    onChange(nextQuery);
+  }
+
   function onReturnKeyDown(e: React.KeyboardEvent<HTMLInputElement>) {
     if (e.key === 'Enter') {
       onRunQuery();
     }
   }
 
   return (
-    <PromQueryField
-      datasource={datasource}
-      query={query}
-      onRunQuery={onRunQuery}
-      onChange={onChange}
-      onBlur={() => {}}
-      history={history}
-      data={data}
-      ExtraFieldElement={
-        <PromExploreExtraField
-          label={'Step'}
-          onChangeFunc={onStepChange}
-          onKeyDownFunc={onReturnKeyDown}
-          value={query.interval || ''}
-          hasTooltip={true}
-          tooltipContent={
-            'Time units can be used here, for example: 5s, 1m, 3h, 1d, 1y (Default if no unit is specified: s)'
-          }
-        />
-      }
-    />
+    <>
+      <PromQueryField
+        datasource={datasource}
+        query={query}
+        onRunQuery={onRunQuery}
+        onChange={onChange}
+        onBlur={() => {}}
+        history={history}
+        data={data}
+        ExtraFieldElement={
+          <PromExploreExtraField
+            label={'Step'}
+            onChangeFunc={onStepChange}
+            onKeyDownFunc={onReturnKeyDown}
+            value={query.interval || ''}
+            hasTooltip={true}
+            tooltipContent={
+              'Time units can be used here, for example: 5s, 1m, 3h, 1d, 1y (Default if no unit is specified: s)'
+            }
+          />
+        }
+      />
+      <PromExploreRadioButton
+        selected={query.runAll ? 'all' : query.instant ? 'instant' : 'range'}
+        onQueryTypeChange={onQueryTypeChange}
+      />
+    </>
+  );
+};
+
+type PromExploreRadioButtonProps = {
+  selected: string;
+  onQueryTypeChange: (value: string) => void;
+};
+
+const PromExploreRadioButton: React.FunctionComponent<PromExploreRadioButtonProps> = ({
+  selected,
+  onQueryTypeChange,
+}) => {
+  const rangeOptions = [
+    { value: 'range', label: 'Range' },
+    { value: 'instant', label: 'Instant' },
+    { value: 'all', label: 'All' },
+  ];
+
+  return (
+    <div
```

Comment:
We are not using label because of the current query editor styling - mixing old and new styles doesn't look that good. This could be updated when we switch to the new styling for the whole prometheus editor.